.github/workflows/pr.yml

```diff
@@ -30,7 +30,17 @@ jobs:
     - name: Check out code
       uses: actions/checkout@v3
 
-    - name: Set go_version variable
+    - name: Set GOPROXY
```

Comment:
I am curious about this change(setup go stuff) ,  is it a must change here ?   
is it possible that this change make the golint-ci crash ? 
if it is not a must change,  I'd suggest just change the above go_version and golangci_version and take a try.

Reply:
it is a must change as we need go 1.21 installed. If we don't have this stage then go1.17 is used. I don't know how this go version is installed by default in github action. I we can change this default version to 1.21 then we can skip this step, however I don't know how to do it. And yes, this change is causing issue with golint-ci.

Reply:
also I take a quick google,  found this https://github.com/edoardottt/boggart/pull/91,  you may take a look and see if it is related.

Reply:
I tried that and then got another issue that "compile: version "go1.17.7" does not match go tool version "go1.21.3". I think we have to figure out why by default we have go 1.17 installed.

Reply:
yes, we need to figure out this.  I take a look at the pr.yaml file,  it looks these stuff are loaded from cache.  take a blame to see this file's history change,  I see initially golanglint-ci was installed explicitly and then removed and use this cache.  so I think we need to look into this cache actions to see how it worked in our case.  
```
    - name: Load build and linter cache
      uses: actions/cache@v2
      with:
        path: |
          ~/.cache/go-build
          ~/.cache/golangci-lint
        key: go-build-${{ env.go_version }}-${{ hashFiles('**/go.sum') }}-${{ hashFiles('**/pkg') }}
        restore-keys: |
          go-build-${{ env.go_version }}-${{ hashFiles('**/go.sum') }}
          go-build-${{ env.go_version }}-
```

Reply:
take further checking,  I think these cache are from this container image(image: ghcr.io/dell/csi-baremetal/csi-baremetal-devkit:latest),  we need to upgrade it.  the golang version and golangci-lint version are determined by this. that's the reason you compilation reported "go 1.17" error. 

```
jobs:
  lint-test:
    name: Validate code and build it
    runs-on: ubuntu-22.04
    container:
      image: ghcr.io/dell/csi-baremetal/csi-baremetal-devkit:latest
```

Reply:
That makes sense. Let me try this.